ParseVariables function parses regular expressions with escape characters incorrectly

code:
```
func TestParseRule(t *testing.T) {
	tests := []struct {
		name string
		vars string
		want int
	}{
		{"Does not contain escape characters", `ARGS:/(test)/|REQUEST_XML`, 2},
		{"The last variable contains escape characters", `ARGS|REQUEST_XML:/(test)\b/`, 2},
		{"Contains escape characters", `ARGS:/(test\b)/|REQUEST_XML`, 2},
	}

	for _, tc := range tests {
		tt := tc
		t.Run(tt.name, func(t *testing.T) {
			rp := RuleParser{
				rule: corazawaf.NewRule(),
			}
			if err := rp.ParseVariables(tt.vars); err != nil {
				t.Error(err)
			}
			for i := 0; i < reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Len(); i++ {
				fmt.Println(i, reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Index(i).FieldByName("KeyStr").String())
			}
			got := reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Len()
			if got != tt.want {
				t.Error("variables parse error want", tt.want, "got", got)
			}
		})
	}
}
```

output:
```
=== RUN   TestParseRule
=== RUN   TestParseRule/Does_not_contain_escape_characters
0 /(test)/
1 
=== RUN   TestParseRule/The_last_variable_contains_escape_characters
0 
1 /(test)\b/
=== RUN   TestParseRule/Contains_escape_characters
0 /(test\b)/|REQUEST_XM/
    rule_parser_test.go:305: variables parse error want 2 got 1
--- FAIL: TestParseRule (0.00s)
    --- PASS: TestParseRule/Does_not_contain_escape_characters (0.00s)
    --- PASS: TestParseRule/The_last_variable_contains_escape_characters (0.00s)
    --- FAIL: TestParseRule/Contains_escape_characters (0.00s)

```

When my parameter is ARGS:/(test\b)/|REQUEST_XML, I want to get ARGS:/(test\b)/ and REQUEST_XML, but the result is /(test\b)/|REQUEST_XM/

--- a/internal/seclang/rule_parser.go
+++ b/internal/seclang/rule_parser.go
@@ -145,6 +145,9 @@ func (rp *RuleParser) ParseVariables(vars string) error {
 				isEscaped = !isEscaped
 			default:
 				curKey = append(curKey, c)
+				if isEscaped {
+					isEscaped = false
+				}
 			}
 		case 3:
 			// XPATH

--- a/internal/seclang/rule_parser_test.go
+++ b/internal/seclang/rule_parser_test.go
@@ -295,9 +295,6 @@ func TestParseRule(t *testing.T) {
 			if err := rp.ParseVariables(tt.vars); err != nil {
 				t.Error(err)
 			}
-			//for i := 0; i < reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Len(); i++ {
-			//	fmt.Println(i, reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Index(i).FieldByName("KeyStr").String())
-			//}
 			got := reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Len()
 			if got != tt.want {
 				t.Error("variables parse error want", tt.want, "got", got)

--- a/internal/seclang/rule_parser_test.go
+++ b/internal/seclang/rule_parser_test.go
@@ -5,6 +5,7 @@ package seclang
 
 import (
 	"errors"
+	"reflect"
 	"strings"
 	"testing"
 
@@ -271,6 +272,37 @@ func TestRawChainedRules(t *testing.T) {
 		if !strings.HasPrefix(r, "SecRule REQUEST_URI ") {
 			t.Errorf("unexpected rule at line %d: %s", i, r)
 		}
+	}
+}
+
+func TestParseRule(t *testing.T) {
+	tests := []struct {
+		name string
+		vars string
+		want int
+	}{
+		{"Does not contain escape characters", `ARGS_GET:/(test)/|REQUEST_XML`, 2},
+		{"The last variable contains escape characters", `ARGS_GET|REQUEST_XML:/(test)\b/`, 2},
+		{"Contains escape characters", `ARGS_GET:/(test\b)/|REQUEST_XML`, 2},
+	}
+
+	for _, tc := range tests {
+		tt := tc
+		t.Run(tt.name, func(t *testing.T) {
+			rp := RuleParser{
+				rule: corazawaf.NewRule(),
+			}
+			if err := rp.ParseVariables(tt.vars); err != nil {
+				t.Error(err)
+			}
+			//for i := 0; i < reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Len(); i++ {
+			//	fmt.Println(i, reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Index(i).FieldByName("KeyStr").String())
+			//}
+			got := reflect.ValueOf(rp.rule).Elem().FieldByName("variables").Len()
+			if got != tt.want {
+				t.Error("variables parse error want", tt.want, "got", got)
+			}
+		})
 	}
 }
 

